pkg/cmd/cli/cmd/login.go

```diff
@@ -75,16 +115,60 @@ func NewCmdLogin(f *osclientcmd.Factory, reader io.Reader, out io.Writer) *cobra
 	return cmds
 }
 
-// RunLogin contains all the necessary functionality for the OpenShift cli login command
-func RunLogin(cmd *cobra.Command, options *LoginOptions) error {
-	if certFile := cmdutil.GetFlagString(cmd, "client-certificate"); len(certFile) > 0 {
-		options.CertFile = certFile
+func (o *LoginOptions) Complete(f *osclientcmd.Factory, cmd *cobra.Command, args []string) error {
+	kubeconfig, err := f.OpenShiftClientConfig.RawConfig()
+	if err != nil {
+		return err
+	}
+	o.StartingKubeConfig = &kubeconfig
+
+	if len(args) == 1 {
+		addr := flagtypes.Addr{Value: "localhost:8443", DefaultScheme: "https", DefaultPort: 8443, AllowPrefix: true}.Default()
+		if err := addr.Set(args[0]); err != nil {
+			return err
+		}
+		o.Server = addr.String()
+
+	} else {
+		if len(o.Server) == 0 {
+			if context, exists := o.StartingKubeConfig.Contexts[o.StartingKubeConfig.CurrentContext]; exists {
+				if cluster, exists := o.StartingKubeConfig.Clusters[context.Cluster]; exists {
+					o.Server = cluster.Server
+				}
+			}
+		}
+	}
+
+	if certFile := kcmdutil.GetFlagString(cmd, "client-certificate"); len(certFile) > 0 {
+		o.CertFile = certFile
 	}
-	if keyFile := cmdutil.GetFlagString(cmd, "client-key"); len(keyFile) > 0 {
-		options.KeyFile = keyFile
+	if keyFile := kcmdutil.GetFlagString(cmd, "client-key"); len(keyFile) > 0 {
+		o.KeyFile = keyFile
 	}
-	options.PathToSaveConfig = cmdutil.GetFlagString(cmd, config.OpenShiftConfigFlagName)
+	o.PathToSaveConfig = kcmdutil.GetFlagString(cmd, config.OpenShiftConfigFlagName)
 
+	o.CertFile = kcmdutil.GetFlagString(cmd, "certificate-authority")
```

Comment:
Nope, that was a bug.  Just a consistent one that I didn't catch because I didn't authenticate using client certs.